rules/CodingStyle/NodeFactory/JsonEncodeStaticCallFactory.php

```diff
@@ -34,7 +36,11 @@ public function createFromArray(Expr $assignExpr, Array_ $jsonArray): Assign
         $jsonDataAssign = new Assign($assignExpr, $jsonArray);
 
         $jsonDataVariable = new Variable('jsonData');
-        $jsonDataAssign->expr = $this->nodeFactory->createStaticCall('Nette\Utils\Json', 'encode', [$jsonDataVariable]);
+        if ($this->reflectionProvider->hasClass('Nette\Utils\Json')) {
+            $jsonDataAssign->expr = $this->nodeFactory->createStaticCall('Nette\Utils\Json', 'encode', [$jsonDataVariable]);
+        } else {
+            $jsonDataAssign->expr = $this->nodeFactory->createFuncCall('json_encode', [$jsonDataVariable]);
+        }
```

Comment:
Thank you for the PR :+1:

The name of class `JsonEncodeStaticCallFactory`  suggests this must create a `StaticCall`. Anyting else is unexpected.

This should be resolved in the rule level:

https://github.com/rectorphp/rector/blob/d3c644f12fa572c6a4d047c18113f0a9b516a6f2/rules/CodingStyle/Rector/Assign/ManualJsonStringToJsonEncodeArrayRector.php#L111

Reply:
Moved to the `ManualJsonStringToJsonEncodeArrayRector`